### PR TITLE
Refactor: Clean up file structure, improve theme switching for diagrams, and separate logic for better modularity

### PR DESCRIPTION
**Pull request by connellr023**
This PR includes several improvements and refactoring to enhance modularity and readability across the codebase:
- Reorganized and removed unnecessary files to streamline the project structure.
- Reworked theme switching for diagrams, allowing colors to adjust dynamically based on light or dark mode for improved readability.
- Enabled different colors for diagrams in light and dark themes.
- Separated logic for timestamps, and stores into distinct modules for better maintainability.
- Moved modal and PocketBase related logic into their own custom hooks.

**Note:** Further testing is still required to ensure everything still works as expected.

### CODE REVIEW

**Review by EtSubas — COMMENTED** [review]
looks good as long as youve tested that all the commands are written correctly and all the subscribes are still working as before, looks like the dumpy you have would provide that part.

File: RocketControlUnitGUI/src/lib/hooks/useModals.ts [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+		};
+
+		modalStore.trigger(modal);
+	};
+
+	const instantStateChange = (state: string) => {
+		nextStatePending = state;
+		pocketbaseHook.writeStateChange(nextStatePending);
+		nextStatePending = '';
+	};
```
Comment (Jgerbrandt) [inline RocketControlUnitGUI/src/lib/hooks/useModals.ts:53]: This one isn't technically a modal, should this stay here?

File: RocketControlUnitGUI/src/routes/+page.svelte [hunk 2]
```diff
@@ -276,366 +188,77 @@
...
-	: ($timer_remaining / 1000).toFixed(0); // Convert to seconds
+	$: timer_period_display = $timer_period === undefined ? 'N/A' : ($timer_period / 1000).toFixed(0); // Convert to seconds
+	$: timer_remaining_display = $timer_remaining === undefined ? 'N/A' : ($timer_remaining / 1000).toFixed(0); // Convert to seconds
 
 	$: relayStatusOutdated = Date.now() - timestamps.relay_status > 5000;
 	$: combustionControlStatusOutdated = Date.now() - timestamps.combustion_control_status > 5000;
 	$: rcuTempOutdated = Date.now() - timestamps.rcu_temp > 5000;
 	$: batteryOutdated = Date.now() - timestamps.battery > 5000;
-	$: dmbPressureOutdated = Date.now() - timestamps.dmb_pressure > 5000;
+	// $: dmbPressureOutdated = Date.now() - timestamps.dmb_pressure > 5000; // This does not seem to be used
```
[the remark this answers is not in the document]
Comment (connellr023) [reply RocketControlUnitGUI/src/routes/+page.svelte]: It is removed now

File: RocketControlUnitGUI/src/lib/hooks/useModals.ts (hunk 1 above)
Comment (connellr023) [reply RocketControlUnitGUI/src/lib/hooks/useModals.ts]: It is not, but it needs the internal state of the hook. I renamed the hook to useInteraction.

File: RocketControlUnitGUI/src/store.ts
[the remark this answers is not in the document]
Comment (connellr023) [reply RocketControlUnitGUI/src/store.ts:1]: I combined them.

File: RocketControlUnitGUI/src/lib/hooks/usePocketbase.ts [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+			command: state
+		});
+	};
+
+	const writeArbitraryCommand = async (target: string, command: string) => {
+		await pocketbase.collection('CommandMessage').create({
+			target,
+			command
+		});
+	};
```
[the remark this answers is not in the document]
Comment (connellr023) [reply RocketControlUnitGUI/src/lib/hooks/usePocketbase.ts]: Done

**Review by Jgerbrandt — APPROVED** [review]

**Comment by EtSubas** [comment]
As long as youve tested everything looks good to me

**Review by EtSubas — APPROVED** [review]

**Pull request merged**